.yarn/patches/@metamask-keyring-controller-npm-15.0.0-fa070ce311.patch

```diff
@@ -0,0 +1,120 @@
+diff --git a/dist/chunk-52QZQQKP.mjs b/dist/chunk-52QZQQKP.mjs
+index 934f432c8013a6af5303726e1495bed2335fa078..8de2560fe81dfc3dbfef83dd0079482306c425d9 100644
+--- a/dist/chunk-52QZQQKP.mjs
++++ b/dist/chunk-52QZQQKP.mjs
+@@ -2,7 +2,8 @@ import {
+   __privateAdd,
+   __privateGet,
+   __privateMethod,
+-  __privateSet
++  __privateSet,
++  KeyringControllerError
+ } from "./chunk-NAAWD7HX.mjs";
+ 
+ // src/KeyringController.ts
+@@ -582,6 +583,18 @@ var KeyringController = class extends BaseController {
+       })
+     );
+     serializedKeyrings.push(...__privateGet(this, _unsupportedKeyrings));
++    /**
++     * ============================== PATCH INFORMATION ==============================
++     * The HD keyring is the default keyring for all wallets if this keyring is missing
++     * for some reason we should avoid saving the keyrings
++     *
++     * The upstream fix is here: https://github.com/MetaMask/core/pull/4168
++     *
++     * This patch can be found on the core branch `extension-keyring-controller-v13-patch`
++     */
++    if (!serializedKeyrings.some((keyring) => keyring.type === KeyringTypes.hd)) {
++      throw new Error(KeyringControllerError.NoHdKeyring);
++    }
```

Comment:
I have two questions about the patch,

1. Are you adding anything new?
2. Have you checked if everything inside the patch is still necessary?